tests/unit/connector/test_opensearch_output.py

```diff
@@ -360,3 +366,27 @@ def test_message_backlog_is_cleared_after_it_was_written(self):
         self.object._config.message_backlog_size = 1
         self.object.store({"event": "test_event"})
         assert len(self.object._message_backlog) == 0
+
+    @pytest.mark.skip(reason="This test is only for local debugging")
+    def test_opensearch_parallel_bulk(self):
+        config = {
+            "type": "opensearch_output",
+            "hosts": ["localhost:9200"],
+            "default_index": "default_index",
+            "error_index": "error_index",
+            "message_backlog_size": 1,
+            "timeout": 5000,
+        }
+        output: OpensearchOutput = Factory.create({"opensearch_output": config}, mock.MagicMock())
+        uuid_str = str(uuid.uuid4())
+        result = output._search_context.search(
+            index="defaultindex", body={"query": {"match": {"foo": uuid_str}}}
+        )
+        len_before = len(result["hits"]["hits"])
+        output._message_backlog = [{"foo": uuid_str, "_index": "defaultindex"}]
+        output._write_backlog()
+        time.sleep(1)
+        result = output._search_context.search(
+            index="defaultindex", body={"query": {"match": {"foo": uuid_str}}}
+        )
+        assert len(result["hits"]["hits"]) > len_before
```

Comment:
If you'd turn it into a proper test we could also query for a specific document with a timestamp.

Reply:
?

Reply:
The idea would be that you create an event with a certain ID (randomized) and then you query for exactly this ID. That way you know that the write call did actually work. 
But anyway, it's not important if we don't turn it into a proper test.